sdk/core/core-http/lib/pipelineOptions.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { HttpClient } from "./httpClient";
+import { ServiceClientOptions } from "./serviceClient";
+import { RetryOptions } from './policies/exponentialRetryPolicy';
+import { KeepAliveOptions } from './policies/keepAlivePolicy';
+import { RedirectOptions } from './policies/redirectPolicy';
+import { ProxyOptions } from './policies/proxyPolicy';
+import { UserAgentOptions } from './policies/userAgentPolicy';
+
+/**
+ * Returns true if the passed object looks like a PipelineOptions.
+ */
+export function isPipelineOptions(
+  pipelineOrOptions: ServiceClientOptions | PipelineOptions
+): pipelineOrOptions is PipelineOptions {
+  // An empty object is considered options
+  function isEmptyObject(obj: ServiceClientOptions | PipelineOptions) {
+    return Object.keys(obj).length === 0 && obj.constructor === Object;
+  }
+  const options = pipelineOrOptions as PipelineOptions;
+  return (
+    isEmptyObject(pipelineOrOptions) ||
+    !!(options.retryOptions || options.proxyOptions || options.redirectOptions || options.keepAliveOptions || options.userAgentOptions || options.httpClient)
+  );
+}
+
+/**
+ * Defines options that are used to configure the HTTP pieline for
+ * an SDK client.
+ */
+export interface PipelineOptions {
+  /**
+   * The HttpClient implementation to use for outgoing HTTP requests.  Defaults
+   * to DefaultHttpClient.
+   */
+  httpClient?: HttpClient;
+
+  /**
+   * Options that control how to retry failed requests.
+   */
+  retryOptions?: RetryOptions;
+
+  /**
+   * Options to configure a proxy for outgoing requests.
+   */
+  proxyOptions?: ProxyOptions;
```

Comment:
Storage uses plain `proxy` here, but `retryOptions` and `keepAliveOptions`. `proxyOptions` seems better. Can we align storage as part of this PR? Else I can write a follow-up PR.

Reply:
We also have `ProxySettings` which follows the same shape as the new `ProxyOptions`...

Reply:
Storage allows users to provide the proxy url instead of the separate proxy options. Is that something we want to support too?

Reply:
@bterlson We decided to do the storage work as part of a separate PR this week since it'll need extra review by the China team.  That PR will definitely contain this consistency change!

@ramya-rao-a Accepting a full proxy URL sounds useful, but I don't know that I'd overload the `proxyOptions` field with the `| string` typing; maybe we can tweak `ProxyOptions` to support that?

Reply:
We can add the url to the `ProxyOptions` with the expectation that if url is provided, other options like host, port etc will be ignored

Reply:
We dont need to discuss the url option as part of this PR. We can talk about it when we touch storage after this PR is merged.